dashboard/src/reducers/endpointReducer.js

```diff
@@ -0,0 +1,20 @@
+import { SET_ENDPOINTS } from "../actions/types";
+
+const initialState = {
+  endpoints: {},
+};
+
+const EndpointReducer = (state = initialState, action = {}) => {
+  const { type, payload } = action;
+  switch (type) {
+    case SET_ENDPOINTS:
+      return {
+        ...state,
+        endpoints: Object.assign({}, payload),
+      };
+    default:
+      return state;
+  }
+};
```

Comment:
Are you expecting to add more cases to this `switch` statement?  Otherwise, I think we would be better served by an `if` statement, or maybe even a ternary expression.